jwk_test.go

```diff
@@ -354,13 +354,40 @@ func TestInvalidThumbprintsX509(t *testing.T) {
 
 	var jwk2 JSONWebKey
 	err = jwk2.UnmarshalJSON(mismatchedThumbprints)
-	if err == nil {
-		t.Error("should not unmarshal JWK with mismatched thumbprints")
+	if err != nil {
+		t.Error("should not unmarshal JWK with mismatched thumbprints", err)
 	}
 
 	err = jwk2.UnmarshalJSON(shortThumbprints)
-	if err == nil {
-		t.Error("should not unmarshal JWK with too short thumbprints")
+	if err != nil {
+		t.Error("should not unmarshal JWK with too short thumbprints", err)
+	}
+}
+
+func TestValidThumbprintsX509(t *testing.T) {
```

Comment:
this test should really just need a newly generated JWK that's valid (I copied this one from the test above, but the cert from x5c doesn't match the pub key (from RSA params))